### PR TITLE
ReactDataGrid -> DataGrid, remove class root wrapper

### PR DESCRIPTION
**Pull request by MayhemYDG**
- I've renamed `ReactDataGrid` (the component and file) to `DataGrid`. This makes more sense, we don't import `react-select` as `ReactSelect` for example, or some button components as `ReactButton`.
- I've removed the root component class wrapper. Since we figured out a way to work around the forwardRef types issue when using generics, the class isn't needed anymore.

### CODE REVIEW

File: packages/react-data-grid-addons/src/draggable-header/__tests__/DraggableContainer.spec.js [hunk 1]
```diff
@@ -2,11 +2,11 @@ import React from 'react';
...
 
 import DraggableContainer from '../DraggableContainer';
-import ReactDataGrid from 'react-data-grid';
+import DataGrid from 'react-data-grid';
 
 describe('<DraggableContainer />', () => {
   it('should render grid wrapper to be used as Drag and Drop context with grid as a child component inside', () => {
     const wrapper = shallow(<DraggableContainer />);
-    expect(wrapper.find(ReactDataGrid));
+    expect(wrapper.find(DataGrid));
```
Comment (qili26) [inline packages/react-data-grid-addons/src/draggable-header/__tests__/DraggableContainer.spec.js]: Don't know what this test is for....

File: packages/react-data-grid-addons/src/menu/ContextMenu.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { ContextMenu } from 'react-contextmenu';
 
-export default function ReactDataGridContextMenu({ children }: { children: React.ReactNode }) {
-  return <ContextMenu id="reactDataGridContextMenu">{children}</ContextMenu>;
+export default function DataGridContextMenu({ children }: { children: React.ReactNode }) {
+  return <ContextMenu id="dataGridContextMenu">{children}</ContextMenu>;
```
Comment (qili26) [inline packages/react-data-grid-addons/src/menu/ContextMenu.tsx]: This seems a break change, maybe for `id` we keep it?

Comment (MayhemYDG) [reply packages/react-data-grid-addons/src/menu/ContextMenu.tsx]: Reverted.

File: packages/react-data-grid-addons/src/draggable-header/__tests__/DraggableContainer.spec.js (hunk 1 above)
Comment (MayhemYDG) [reply packages/react-data-grid-addons/src/draggable-header/__tests__/DraggableContainer.spec.js]: It doesn't even have a `toBeDefined()` or anything, so I've removed this test file and another useless one.
The tests are in a dire state right now, and we'll need to overhaul them, but not in this PR.

Comment (qili26) [reply packages/react-data-grid-addons/src/draggable-header/__tests__/DraggableContainer.spec.js]: Agree. This reminds me a jest eslint rule that to always require assertion.

**Pull request merged**